### PR TITLE
PP-12354 Add get Stripe account by serviceId

### PR DESCRIPTION
**Pull request by james-peacock-ho**
- add endpoint to get Stripe account by serviceId and accountType
- add integration tests

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java [hunk 1]
```diff
@@ -54,4 +55,23 @@ public StripeAccountResponse getStripeAccount(
...
+                            content = @Content(schema = @Schema(implementation = StripeAccountResponse.class))),
+                    @ApiResponse(responseCode = "404", description = "Not found - Service does not exist or service does not have a stripe gateway account of this type")
+            }
+    )
+    public StripeAccountResponse getStripeAccountByServiceIdAndAccountType(
+            @PathParam("serviceId") String serviceId, @PathParam("accountType") GatewayAccountType accountType) {
+        return gatewayAccountService.getGatewayAccountByServiceIdAndAccountType(serviceId, accountType)
+                .filter(StripeAccountUtils::isStripeGatewayAccount)
+                .flatMap(stripeAccountService::buildStripeAccountResponse)
+                .orElseThrow(NotFoundException::new);
```
Comment (DomBelcher) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: We could add in specific error messages for if no account is found, or the account is not a Stripe account, or there are no stripe credentials

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java [hunk 2]
```diff
@@ -2,67 +2,120 @@
...
 
-        app.givenSetup()
-                .get("/v1/api/accounts/" + testAccount.getAccountId() + "/stripe-account")
-                .then()
-                .statusCode(200)
-                .body("stripe_account_id", is(STRIPE_ACCOUNT_ID));
+    @Nested
+    class GetStripeAccountByGatewayAccountId {
+        @Test
+        public void getStripeAccountReturnsSuccessfulResponse() {
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java]: Could remove `public` modifier and `getStripeAccount` from method name, as this is included in the nested class name - similarly for other tests
While these are existing tests, it's only a minor change that should improve readability

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java [hunk 3]
```diff
@@ -2,67 +2,120 @@
...
+                    .get("/v1/api/accounts/" + testAccount.getAccountId() + "/stripe-account")
+                    .then()
+                    .statusCode(404);
+        }
     }
+    
+    @Nested
+    class GetStripeAccountByServiceIdAndAccountType {
+        @Test
+        public void getStripeAccountReturnsSuccessfulResponse() {
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java]: As above, could shorten to `returnsSuccessfulResponse`, and remove `public`

**Review by DomBelcher — CHANGES_REQUESTED** [review]
Nice work, just a few small suggestions

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java [hunk 4]
```diff
@@ -1,68 +1,125 @@
...
-                .withPaymentProvider("sandbox")
-                .withCredentials(Collections.singletonMap("stripe_account_id", STRIPE_ACCOUNT_ID))
-                .insert();
+        @Test
+        void returnsNotFoundResponseWhenServiceIdNotFound() {
+            app.givenSetup()
+                    .get("/v1/api/service/unknown-service-id/TEST/stripe-account")
+                    .then()
+                    .statusCode(404)
+                    .body("message[0]", is(format("Gateway account not found for service ID [unknown-service-id] and account type [test]")));
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java]: `format` isn't necessary if hardcoding the `serviceId` and `accountType`

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java [hunk 5]
```diff
@@ -1,68 +1,125 @@
...
+    
+    @Nested
+    class GetStripeAccountByServiceIdAndAccountType {
+        @Test
+        void returnsSuccessfulResponse() {
+            String accountId = testBaseExtension.createAGatewayAccountWithServiceId("a-valid-service-id", "stripe");
+            Map<String, String> credentials = Map.of("stripe_account_id", STRIPE_ACCOUNT_ID);
+            app.givenSetup()
+                    .body(toJson(Map.of("payment_provider", "stripe", "credentials", credentials)))
+                    .post("/v1/api/accounts/" + accountId + "/credentials");
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceIT.java]: Might as well add a `.then().statusCode(200)` here

**Review by DomBelcher — CHANGES_REQUESTED** [review]
Nice work, just a couple of very minor changes

**Comment by oswaldquek** [comment]
Separation between the `@Nested` GetStripeAccountByGatewayAccountId and GetStripeAccountByServiceIdAndAccountType - nice! 🎉

**Review by DomBelcher — APPROVED** [review]
Nice work 👍

**Pull request merged**